lib/track.ts

```diff
@@ -128,4 +135,33 @@ export class TrackClient {
       `${this.trackRoot}/customers/${encodeURIComponent(customerId)}/devices/${encodeURIComponent(deviceToken)}`,
     );
   }
+
+  isValidIdType(input: string) {
+    return ["id", "email", "cio_id"].includes(input)
+  }
+
+  mergeCustomers(primaryIdType: string, primaryId: string | number, secondaryIdType: string, secondaryId: string | number) {
```

Comment:
If you make the `Identifier` enum, you can update the types for `primaryIdType` and `secondaryIdType` to be `Identifier`:

```suggestion
  mergeCustomers(primaryIdType: Identifier, primaryId: string | number, secondaryIdType: Identifier, secondaryId: string | number) {
```

The checks on lines 145 and 152 would need to remain for non-typescript clients. But, for the users who use this project with Typescript, they can get an error earlier during compile time, which is better.